TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/fsm/InsaneStateMachineException.java

```diff
@@ -0,0 +1,10 @@
+package org.firstinspires.ftc.teamcode.subsystems.fsm;
+
+/**
+ * A runtime exception representing an insane state machine configuration encountered at runtime.
+ */
+public class InsaneStateMachineException extends RuntimeException {
+    public InsaneStateMachineException(String reason) {
+        super("Encountered an insane state machine: " + reason + ".");
```

Comment:
Totally! Since this class extends `RuntimeException`, the `super` in this constructor calls the matching constructor in the `RuntimeException` abstract class. It's sorta like:

```java
this = new RuntimeException("Encountered an insane state machine: " + reason + ".");
```

But actually valid.